velox/functions/prestosql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -341,6 +341,48 @@ TEST_F(AverageAggregationTest, avgDecimal) {
       {"avg(c0)"},
       {},
       {makeRowVector({makeShortDecimalFlatVector({3'000}, DECIMAL(10, 1))})});
+
+  // Decimal average aggregation with multiple groups.
+  auto inputRows = {
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({1, 1}),
+           makeShortDecimalFlatVector({37220, 53450}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({2, 2}),
+           makeShortDecimalFlatVector({10410, 9250}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({3, 3}),
+           makeShortDecimalFlatVector({-12783, 0}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({1, 2}),
+           makeShortDecimalFlatVector({23178, 41093}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({2, 3}),
+           makeShortDecimalFlatVector({-10023, 5290}, DECIMAL(5, 2))}),
+  };
+
+  auto expectedResult = {
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({1}),
+           makeShortDecimalFlatVector({37949}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({2}),
+           makeShortDecimalFlatVector({12683}, DECIMAL(5, 2))}),
+      makeRowVector(
+          {makeNullableFlatVector<int32_t>({3}),
+           makeShortDecimalFlatVector({-2498}, DECIMAL(5, 2))})};
+
+  testAggregations(inputRows, {"c0"}, {"avg(c1)"}, expectedResult);
+}
+
+TEST_F(AverageAggregationTest, avgDecimalWithMultipleRowVectors) {
+  auto inputRows = {
+      makeRowVector({makeShortDecimalFlatVector({100, 200}, DECIMAL(5, 2))}),
```

Comment:
Weird, how come this test passed?  They should be 100 time smaller than the expected values.

Reply:
It is a bug in Velox. It interprets literals as unscaled values. I replaced the expected result with an actual decimal answer. I will follow up on this.

Reply:
This was actually fixed on Friday: https://github.com/facebookincubator/velox/commit/b1ba9b536344bd41e59789a6aeac74e3d249526e so I did not rebase since then. `decimalVariantAt` was replaced with the function from DuckDB so it would return the correct result, what you would expect.

I also tested on the latest master and this bug does not exist anymore. I suggest we keep the code as is, just compare it against another, expected, vector.